src/plugins/discourse/references.js

```diff
@@ -2,8 +2,33 @@
 
 const htmlparser2 = require("htmlparser2");
 
-export type Hyperlink = string;
+import {type TopicId} from "./fetch";
+
+export type DiscoursePostReference = {|
+  +type: "POST",
+  +topicId: TopicId,
+  +postIndex: number,
+  +serverUrl: string,
+|};
+
+export type DiscourseTopicReference = {|
+  +type: "TOPIC",
+  +topicId: TopicId,
+  +serverUrl: string,
+|};
+
+export type DiscourseUserReference = {|
+  +type: "USER",
+  +username: string,
+  +serverUrl: string,
+|};
+
+export type DiscourseReference =
+  | DiscoursePostReference
+  | DiscourseTopicReference
+  | DiscourseUserReference;
 
+export type Hyperlink = string;
```

Comment:
This name suggests to me that the type represents the source for an HTML
or Markdown link, like `"<a>…</a>"` or `[…](…)`. But it seems from
context that these are actually just URL strings. What do you think of
using `UrlString` instead (to avoid confusion with builtin `URL`)?

Reply:
Done.